sqe-database-access/TextRepository.cs

```diff
@@ -325,10 +366,243 @@ UserInfo editionUser
+				{
+					// Create a new dummy attribute for a start break of a scroll
+					attr = SignInterpretationAttributeFactory.CreateElementTerminatorAttributes(
+							TableData.Table.manuscript
+							, TableData.TerminatorType.Start);
+
+					// Add it to the first sign of the new line
+					// since the first sign of a line has by default only one sing inteprretaion
+					// holding all the break attributes we can simply chooose
+					// SignInterpretations[0]
+					newAttributes =
+							await _attributeRepository.CreateSignInterpretationAttributesAsync(
+									editionUser
+									, newLine.Signs[0]
+											 .SignInterpretations[0]
+											 .SignInterpretationId.Value
+									, attr);
+
+					// Add it also to the result
+					newLine.Signs[0].SignInterpretations[0].Attributes.AddRange(newAttributes);
+				}
+
+				return newLine;
+			}
+
+			return null;
+		}
+
+		public async Task<LineData> InsertLineAfterAsync(
+				UserInfo   editionUser
+				, uint     fragmentId
+				, LineData lineData
+				, uint     previouslineId)
+		{
+			// Get the SignInterpretationId of the of the end break of the previous line
+			var anchorBefore =
+					_getTerminators(editionUser, TableData.Table.line, previouslineId).EndId;
+
+			// Try to get the SignInterpretationId of the start break of the next line
+			var anchorsAfter = await _getNextSignInterpretationIds(
+					editionUser.EditionId.GetValueOrDefault()
+					, anchorBefore);
+
+			// If there is no next line than set the anchorAfter to 0
+			var anchorAfter = (anchorsAfter != null) && (anchorsAfter.Count() > 0)
+					? anchorsAfter.First()
+					: 0;
+
+			// Creat the new line with all signs already given in line data.
+			var newLine = await CreateLineAsync(
+					editionUser
+					, lineData
+					, fragmentId
+					, anchorBefore
+					, anchorAfter);
+
+			// If anchorAfter == 0 then the new line is the new last line of thef fragment.
+			// In this case we have to move the end break for the text fragment from the
+			// old last line to the new last line
+			if (anchorAfter == 0)
+			{
+				// Create the dummy break attribute
+				var attr = SignInterpretationAttributeFactory.CreateElementTerminatorAttributes(
+						TableData.Table.text_fragment
+						, TableData.TerminatorType.End);
+
+				// Add it to the last sign of the newline
+				// since the last sign of a line has by default only one sing inteprretaion
+				// holding all the break attributes we can simply chooose
+				// SignInterpretations[0]
+				var newAttributes =
+						await _attributeRepository.CreateSignInterpretationAttributesAsync(
+								editionUser
+								, newLine.Signs.Last()
+										 .SignInterpretations[0]
+										 .SignInterpretationId.Value
+								, attr);
+
+				// Add the attribute also to the return value.
+				newLine.Signs.Last().SignInterpretations[0].Attributes.AddRange(newAttributes);
+
+				// Delete the end break from the old last line
+				//TODO How do we signal this change?
```

Comment:
Perhaps in case of changes that are substantial (here, in the previous function and maybe later as well), we should have a catch all "please query this again" SignalR message, so clients know their sign interpretation data is invalid, and they need to query it again?)

Reply:
Yeah, this method is not yet exposed in the API, but we will need to do something like that.